numba/tests/support.py

```diff
@@ -409,27 +411,63 @@ def tweak_code(func, codestring=None, consts=None):
                       co.co_lnotab)
     func.__code__ = new_code
 
-def static_temp_directory(dirname):
-    """
-    Create a directory in the temp dir with a given name. Statically-named
-    temp dirs created using this function are needed because we can't delete a
-    DLL under Windows (this is a bit fragile if stale files can influence the
-    result of future test runs).
-    """
-    if os.name == 'nt':
-        # Under Windows, gettempdir() points to the user-local temp dir
-        tmpdir = os.path.join(tempfile.gettempdir(), dirname)
-    else:
-        # Mix the UID into the directory name to allow different users to
-        # run the test suite without permission errors (issue #1586)
-        tmpdir = os.path.join(tempfile.gettempdir(),
-                              "%s.%s" % (dirname, os.getuid()))
+
+_trashcan_dir = 'numba-tests'
+
+if os.name == 'nt':
+    # Under Windows, gettempdir() points to the user-local temp dir
+    _trashcan_dir = os.path.join(tempfile.gettempdir(), _trashcan_dir)
+else:
+    # Mix the UID into the directory name to allow different users to
+    # run the test suite without permission errors (issue #1586)
+    _trashcan_dir = os.path.join(tempfile.gettempdir(),
+                                 "%s.%s" % (_trashcan_dir, os.getuid()))
+
+# Stale temporary directories are deleted after they are older than this value.
+# The test suite probably won't ever take longer than this...
+_trashcan_timeout = 24 * 3600
```

Comment:
I gather this is 'one day'. It would be good to spell that out, if only in comment.